samples/scheduling/hangfire/Core_6/Scheduler/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Threading.Tasks;
+using Hangfire;
+using Hangfire.MemoryStorage;
+using NServiceBus;
+using NServiceBus.Logging;
+using NServiceBus.Serilog;
+using Scheduler;
+using Serilog;
+
+class Program
+{
+    static async Task Main()
+    {
+        #region serilog
+
+        Log.Logger = new LoggerConfiguration()
+            .WriteTo.ColoredConsole()
+            .CreateLogger();
+        LogManager.Use<SerilogFactory>();
+
+        #endregion
+
+        Console.Title = "Samples.HangfireScheduler.Scheduler";
+        var endpointConfiguration = new EndpointConfiguration("Samples.HangfireScheduler.Scheduler");
+        endpointConfiguration.UseTransport<LearningTransport>();
+
+        #region Configuration
+
+        var endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+
+        // store the endpointInstance in a static helper class
+        EndpointHelper.Instance = endpointInstance;
+
+        //use in memory storage. Production scenarios should use more robust alternatives: SqlServer, Msmq, Redis etc
+        GlobalConfiguration.Configuration.UseMemoryStorage();
+
+        // create and start scheduler instance
+        var scheduler = new BackgroundJobServer();
+
+        #endregion
+
+        #region scheduleJob
+        
+        // Tell Hangfire to schedule the job and trigger every minute 
+        RecurringJob.AddOrUpdate("myJob", () => SendMessageJob.Run(), Cron.Minutely);
```

Comment:
out of curiosity how does the static `RecurringJob` use the instance `scheduler`?

Reply:
Through some kind of magic. You have the same thing with configuration. You have a static `GlobalConfiguration` class that you use to configure and that is used behind the cover when you do `var scheduler = new BackgroundJobServer();`

Reply:
hmm. weird. but it is actually what i saw in some snippets, so dont doubt u. just seems to be asking for some side effects. not that this is our problem given the context of this sample. 

thanks for the clarification